src/anomalib/models/uflow/lightning_model.py

```diff
@@ -0,0 +1,120 @@
+    ) -> None:
+        super().__init__()
+        self.model: UflowModel = UflowModel(
+            input_size=input_size,
+            backbone=backbone,
+            flow_steps=flow_steps,
+            affine_clamp=affine_clamp,
+            affine_subnet_channels_ratio=affine_subnet_channels_ratio,
+            permute_soft=permute_soft,
+        )
+        self.loss = UFlowLoss()
+
+    def training_step(self, batch: dict[str, str | Tensor], *args, **kwargs) -> STEP_OUTPUT:
+        z, ljd = self.model(batch['image'])
+        loss = self.loss(z, ljd)
+        self.log_dict({"loss": loss}, on_step=True, on_epoch=False, prog_bar=False, logger=True)
+        return {"loss": loss}
+
+    def validation_step(self, batch: dict[str, str | Tensor], *args, **kwargs) -> STEP_OUTPUT:
+        anomaly_maps = self.model(batch["image"])
+        batch["anomaly_maps"] = anomaly_maps
+        return batch
+
+
+class UflowLightning(Uflow):
+    """PL Lightning Module for the UFLOW algorithm.
+
+    Args:
+        hparams (DictConfig | ListConfig): Model params
+    """
+
+    def __init__(self, hparams: DictConfig | ListConfig) -> None:
+        super().__init__(
+            input_size=hparams.model.input_size,
+            backbone=hparams.model.backbone,
+            flow_steps=hparams.model.flow_steps,
+            affine_clamp=hparams.model.affine_clamp,
+            affine_subnet_channels_ratio=hparams.model.affine_subnet_channels_ratio,
+            permute_soft=hparams.model.permute_soft
+        )
+        self.lr = hparams.model.lr
+        self.weight_decay = hparams.model.weight_decay
+        self.hparams: DictConfig | ListConfig  # type: ignore
+        self.save_hyperparameters(hparams)
+
+    def configure_callbacks(self) -> list[EarlyStopping]:
+        """Configure model-specific callbacks.
+
+        Note:
+            This method is used for the existing CLI.
+            When PL CLI is introduced, configure callback method will be
+                deprecated, and callbacks will be configured from either
+                config.yaml file or from CLI.
+        """
+        early_stopping = EarlyStopping(
+            monitor=self.hparams.model.early_stopping.metric,
+            patience=self.hparams.model.early_stopping.patience,
+            mode=self.hparams.model.early_stopping.mode,
+        )
+        return [early_stopping]
+
+    def configure_optimizers(self):
+        def get_total_number_of_iterations():
+            try:
+                self.trainer.reset_train_dataloader()
+                number_of_training_examples = len(self.trainer.train_dataloader.dataset)
+                batch_size = self.trainer.train_dataloader.loaders.batch_size
+                drop_last = 1 * self.trainer.train_dataloader.loaders.drop_last
+                iterations_per_epoch = number_of_training_examples // batch_size + 1 - drop_last
+                total_iterations = iterations_per_epoch * (self.trainer.max_epochs - 1)
+            except:
```

Comment:
Honestly, I'm not being able to reproduce this error... At some point (as far as I remember), the error was generated because, at this point, the dataloaders are not yet initialized. It seems like they are initialized in the .fit loop (according to [this](https://github.com/Lightning-AI/lightning/issues/10430#issue-1048444548)).